### PR TITLE
feat(ai): report OpenAI Agents tracing processor errors

### PR DESCRIPTION
**Pull request by marandaneto**
## Problem

The OpenAI Agents tracing processor intentionally avoids throwing from tracing lifecycle hooks, but failures were fully silent. That made capture and flush issues hard to observe and also left defensive-code scan findings around swallowed errors.

## Changes

- Add an optional `onError(error, context)` callback to `PostHogTracingProcessorOptions`.
- Type the error callback context with an exported `TracingProcessorErrorContext` union.
- Route capture, trace/span lifecycle, shutdown, and force flush failures through `onError` while preserving the existing non-throwing behavior.
- Guard the `onError` callback itself so a throwing handler cannot break the processor's non-throwing contract.
- Harden timestamp parsing against invalid runtime values and harden captured-value size checks against JSON serialization failures.
- Reuse shared AI utility helpers for content stringification and UTF-8 byte length checks.
- Export the tracing processor error handler/context types from the OpenAI Agents entrypoint.
- Add tests covering capture, flush, throwing-handler, invalid timestamp, and unserializable payload handling.

### Libraries affected

- [ ] All of them
- [ ] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [ ] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react-native-plugin
- [ ] @posthog/react
- [x] @posthog/ai
- [ ] @posthog/convex
- [ ] @posthog/next
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types
- [ ] @posthog/browser-common

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [x] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [x] Ran `pnpm changeset` to generate a changeset file

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

A pi coding agent ran `slop-scan`, selected the OpenAI Agents tracing processor as a useful low-noise hotspot, and made a focused change to keep the processor's non-throwing behavior while exposing failures through an optional callback. Reviewer feedback and follow-up hardening were addressed by guarding the callback, typing the context, reusing shared utilities, avoiding double serialization preparation, and handling invalid runtime timestamps/serialization edge cases. Validation run: `pnpm --filter=@posthog/ai test:unit -- openai-agents.test.ts --runInBand`, `pnpm --filter=@posthog/ai lint`, `git diff --check`, and `slop-scan scan packages/ai/src/openai-agents --lint`.

### CODE REVIEW

File: packages/ai/src/openai-agents/processor.ts [hunk 1]
```diff
@@ -178,6 +175,10 @@ export class PostHogTracingProcessor implements TracingProcessor {
     }
   }
 
+  private _handleError(error: unknown, context: string): void {
+    this._onError?.(error, context)
+  }
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply packages/ai/src/openai-agents/processor.ts]: Good catch! Fixed in c282bd0b2 —  now guards against throwing  callbacks, preserving the processor's non-throwing contract, with test coverage.

[the remark this answers is not in the document]
Comment (marandaneto) [reply packages/ai/src/openai-agents/processor.ts]: Correction: `_handleError` now guards against throwing `onError` callbacks, preserving the processor non-throwing contract, with test coverage in c282bd0b2.

**Comment by marandaneto** [comment]
@PostHog/team-ai-observability not sure if we want this, but this is a good thing for customers and us as sdk observability at least

File: packages/ai/src/openai-agents/processor.ts [hunk 2]
```diff
@@ -43,41 +43,62 @@ function normalizeInputRoles(input: unknown): unknown {
...
-    return isNaN(ts) ? null : ts / 1000
+    return JSON.stringify(serializableValue) ?? safeString(serializableValue)
   } catch {
+    return safeString(serializableValue)
+  }
+}
+
+function exceedsMaxOutputSize(value: unknown): boolean {
+  const serializedValue = stringifyForSizeCheck(value)
+  return serializedValue === null ? false : new TextEncoder().encode(serializedValue).length > MAX_OUTPUT_SIZE
```
Comment (ioannisj) [inline packages/ai/src/openai-agents/processor.ts]: Could potentially reuse https://github.com/PostHog/posthog-js/blob/e480a3e23ecff45d2f9cf50332f6f59c54a7c736/packages/ai/src/utils.ts#L47?

File: packages/ai/src/openai-agents/processor.ts [hunk 3]
```diff
@@ -89,13 +110,15 @@ interface TraceMetadata {
 }
 
 export type DistinctIdResolver = string | ((trace: Trace) => string | null | undefined)
+export type TracingProcessorErrorHandler = (error: unknown, context: string) => void
```
Comment (ioannisj) [inline packages/ai/src/openai-agents/processor.ts]: should context be typed here?

File: packages/ai/src/openai-agents/processor.ts [hunk 4]
```diff
@@ -43,41 +43,62 @@ function normalizeInputRoles(input: unknown): unknown {
...
+    const serializedValue = JSON.stringify(obj)
+    return serializedValue === undefined ? safeString(obj) : obj
   } catch {
-    return String(obj)
+    return safeString(obj)
   }
 }
 
-function exceedsMaxOutputSize(value: unknown): boolean {
+function stringifyForSizeCheck(value: unknown): string | null {
```
Comment (ioannisj) [inline packages/ai/src/openai-agents/processor.ts:66]: This calls ensureSerializable() but it's being handed an already serialized value from _prepareCapturedValue()

**Review by ioannisj — APPROVED** [review]

File: packages/ai/src/openai-agents/processor.ts (hunk 2 above)
Comment (marandaneto) [reply packages/ai/src/openai-agents/processor.ts]: Done in 9819cf3b4 — reused the shared utilities from `../utils` (`toContentString` and `utf8ByteLength`) instead of keeping the local raw string/encoder path.

File: packages/ai/src/openai-agents/processor.ts (hunk 3 above)
Comment (marandaneto) [reply packages/ai/src/openai-agents/processor.ts]: Done in 9819cf3b4 — added/exported `TracingProcessorErrorContext` as a union of the lifecycle contexts and used it in the handler and `_handleError`.

File: packages/ai/src/openai-agents/processor.ts (hunk 4 above)
Comment (marandaneto) [reply packages/ai/src/openai-agents/processor.ts:66]: Done in 9819cf3b4 — `_prepareCapturedValue` now calls `ensureSerializable` once, then size-checks the already-prepared value without running it through `ensureSerializable` again.
